app/src/main/java/org/mozilla/fenix/downloads/DynamicDownloadDialog.kt

```diff
@@ -0,0 +1,140 @@
+                (layoutParams as CoordinatorLayout.LayoutParams).apply {
+
+                    behavior =
+                        DynamicDownloadDialogBehavior<View>(
+                            context,
+                            null,
+                            toolbarHeight.toFloat()
+                        )
+                }
+            }
+        }
+
+        if (settings.shouldUseBottomToolbar) {
+            val params: ViewGroup.MarginLayoutParams =
+                view.layoutParams as ViewGroup.MarginLayoutParams
+            params.bottomMargin = toolbarHeight
+        }
+
+        if (didFail) {
+            view.download_dialog_title.text =
+                container.context.getString(R.string.mozac_feature_downloads_failed_notification_text2)
+
+            view.download_dialog_icon.setImageResource(
+                mozilla.components.feature.downloads.R.drawable.mozac_feature_download_ic_download_failed
+            )
+
+            view.download_dialog_action_button.apply {
+                text = context.getString(
+                    mozilla.components.feature.downloads.R.string.mozac_feature_downloads_button_try_again
+                )
+                setOnClickListener {
+                    tryAgain(downloadState.id)
+                    context.metrics.track(Event.InAppNotificationDownloadTryAgain)
+                    dismiss(view)
+                }
+            }
+        } else {
+            val titleText = container.context.getString(
+                R.string.mozac_feature_downloads_completed_notification_text2
+            ) + " (${downloadState.contentLength?.toMegabyteString()})"
+
+            view.download_dialog_title.text = titleText
+
+            view.download_dialog_icon.setImageResource(
+                mozilla.components.feature.downloads.R.drawable.mozac_feature_download_ic_download_complete
+            )
+
+            view.download_dialog_action_button.apply {
+                text = context.getString(
+                    mozilla.components.feature.downloads.R.string.mozac_feature_downloads_button_open
+                )
+                setOnClickListener {
+                    val fileWasOpened = AbstractFetchDownloadService.openFile(
+                        context = context,
+                        contentType = downloadState.contentType,
+                        filePath = downloadState.filePath
+                    )
+
+                    if (!fileWasOpened) {
+                        onCannotOpenFile()
+                    }
+
+                    context.metrics.track(Event.InAppNotificationDownloadOpen)
+                    dismiss(view)
+                }
+            }
+        }
+
+        view.download_dialog_close_button.setOnClickListener {
+            dismiss(view)
+        }
+
+        view.download_dialog_filename.text = downloadState.fileName
+    }
+
+    fun expand() {
```

Comment:
Should we rename it to `show` instead of `expand`? Show  feels more familiar as it is normally use for dialog to display its content [see](https://developer.android.com/reference/androidx/fragment/app/DialogFragment#show(androidx.fragment.app.FragmentManager,%20java.lang.String))  What do you think?

Reply:
Yep, makes sense. I'll rename it.